packages/opal-common/opal_common/git/bundle_maker.py

```diff
@@ -53,19 +53,61 @@ def _get_explicit_manifest(self, viewer: CommitViewer) -> Optional[List[str]]:
         loaded into OPA.
 
         This method searches for an explicit manifest file, reads it and returns the list of paths, or None if not found.
-        """
-        def is_manifest_file(f: VersionedFile) -> bool:
-            return f.path == Path(self._manifest_filename)
 
-        manifest_files = list(viewer.files(is_manifest_file))
-        if not manifest_files:
-            return None
+        The manifest file can include references to other directories containing a ".manifest" file,
+        those would be recursively expanded to compile the final manifest list.
+        """
+        visited_paths = []
+
+        def _compile_manifest_file(dir: VersionedDirectory, manifest_file_name: str = ".manifest") -> List[str]:
+            explicit_manifest = list()
```

Comment:
I feel like this function would be more readable if it was a generator, `yield`ing values instead of collecting them up into a list and returning it at the end. But this approach works fine, as well.

However, if you want to use a list, a type hint would be nice:
```python
explicit_manifest: List[Path] = []
```

Reply:
Cool idea, but wouldn't it be a bit confusing as a generator since it's also recursive?
I've added the type hint